### DISCUSSION TITLE
patterns/factory-kit/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#79] [Announcements]
# patterns/factory-kit/

Also Known As Abstract-Factory Intent Define a factory of immutable content with separated builder and factory interfaces. Explanation Real-world example Imagine a magical weapon factory that can create any type of weapon wished for. When the factory is unboxed, the master recites the weapon types needed to prepare it. After that, any of those weapon types can be summoned in an instant.

https://java-design-patterns.com/patterns/factory-kit/

**Comment by jojijoy15** [comment]
This pattern seems same as [this one](https://java-design-patterns.com/patterns/abstract-factory/). Is this just a lambda based operation.
